test/cloudtest/pkg/providers/packet/packet_provider.go

```diff
@@ -305,16 +305,17 @@ func (pi *packetInstance) createDevice(devCfg *config.DeviceConfig) (*packngo.De
 	}
 	var device *packngo.Device
 	var response *packngo.Response
-	for {
+	for i := 0; i < len(pi.facilitiesList); i++ {
+		devReq.Facility = []string{pi.facilitiesList[i]}
 		device, response, err = pi.client.Devices.Create(devReq)
 		msg := fmt.Sprintf("HostName=%v\n%v - %v", hostName, response, err)
 		logrus.Infof(fmt.Sprintf("%s-%v", pi.id, msg))
 		pi.manager.AddLog(pi.id, fmt.Sprintf("create-device-%s", devCfg.Name), msg)
-		if err == nil || err != nil && !strings.Contains(err.Error(), "has no provisionable") || len(devReq.Facility) <= 1 {
+		if err == nil || err != nil &&
+			!strings.Contains(err.Error(), "has no provisionable") &&
+			!strings.Contains(err.Error(), "Oh snap, something went wrong") {
```

Comment:
Hmm would it be better to look for the error code and not rely on specific strings? I don't expect these to be as stable as we would like them to be.

Reply:
Error returned as string:
`POST https://api.packet.net/projects/383890d0-f5d1-4de1-881a-4d1ede549d18/devices: 500 Oh snap, something went wrong! We've logged the error and will take a look - please reach out to us if you continue having trouble.`
I found that substrings - the best way to filter that errors.

Also I already created report to the packet and received answer that they took it in work